craft_application/git/_git_repo.py

```diff
@@ -286,6 +338,119 @@ def rename_remote(
                 f"cannot rename '{remote_name}' to '{new_remote_name}'"
             ) from error
 
+    def show_remote(
+        self,
+        *,
+        remote_name_or_url: str,
+        do_not_query_remotes: bool = False,
+        disable_prompting: bool = True,
+    ) -> str:
```

Comment:
Depending on the overflow discussion this may not end up being in the API at all